mscore/pagesettings.cpp

```diff
@@ -270,311 +321,428 @@ void PageSettings::orientationClicked()
+                  return; // nothing has changed
+                  }
+            }
+      if (runCmd)
+            score->startCmd();
+
+      score->undoChangePageNumberOffset(pageOffset->value() - 1); // why isn't this a style??
+      score->undoChangeStylePtrs(psize, odd, even);
+      score->undoChangeStyleVal(Sid::spatium,          prev->spatium());
+      score->undoChangeStyleVal(Sid::pageTwosided,     twosided->isChecked());
+      score->undoChangeStyleVal(Sid::pageSize,         int(psize.id()));
+      score->undoChangeStyleVal(Sid::pageUnits,        pageUnits [int(odd.units())].key());
+      score->undoChangeStyleVal(Sid::pageOrientation,  pageOrient[int(odd.orientation())]);
+      score->undoChangeStyleVal(Sid::pageFullWidth,    odd.widthPoints());
+      score->undoChangeStyleVal(Sid::pageFullHeight,   odd.heightPoints());
+      score->undoChangeStyleVal(Sid::marginOddLeft,    odd.leftMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddRight,   odd.rightMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddTop,     odd.topMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddBottom,  odd.bottomMarginPoints());
+      score->undoChangeStyleVal(Sid::marginEvenTop,    even.topMarginPoints());
+      score->undoChangeStyleVal(Sid::marginEvenBottom, even.bottomMarginPoints());
+
+      // older 3.01 styles
+      score->undoChangeStyleVal(Sid::pageWidth,  odd.widthPoints()  / PPI);
+      score->undoChangeStyleVal(Sid::pageHeight, odd.heightPoints() / PPI);
+      score->undoChangeStyleVal(Sid::pagePrintableWidth, (odd.widthPoints() - odd.leftMarginPoints() - odd.rightMarginPoints()) / PPI);
+      score->undoChangeStyleVal(Sid::pageEvenTopMargin,    even.topMarginPoints()    / PPI);
+      score->undoChangeStyleVal(Sid::pageEvenBottomMargin, even.bottomMarginPoints() / PPI);
+      score->undoChangeStyleVal(Sid::pageEvenLeftMargin,   even.leftMarginPoints()   / PPI);
+      score->undoChangeStyleVal(Sid::pageOddTopMargin,     odd .topMarginPoints()    / PPI);
+      score->undoChangeStyleVal(Sid::pageOddBottomMargin,  odd .bottomMarginPoints() / PPI);
+      score->undoChangeStyleVal(Sid::pageOddLeftMargin,    odd .leftMarginPoints()   / PPI);
+
+      if (runCmd)
+            score->endCmd();
       }
 
 //---------------------------------------------------------
-//   updatePreview
+//   okCancel
 //---------------------------------------------------------
 
-void PageSettings::updatePreview(int val)
+void PageSettings::okCancel(QAbstractButton* b)
       {
-      updateValues();
-      switch(val) {
-            case 0:
-                  preview->score()->doLayout();
+      switch (buttonBox->standardButton(b)) {
+            case QDialogButtonBox::Apply:
+                  applyToScore(cs);
                   break;
-            case 1:
-                  preview->score()->doLayout();
+            case QDialogButtonBox::Ok:
+                  applyToScore(cs); // FALLTHROUGH
```

Comment:
There is no big deal to add two extra strings, but solves possible issues for future development because of fallthrough cases.